concepts/collect/about.md

```diff
@@ -1,3 +1,85 @@
+order of the items. Since iterator adaptors yield another iterator, they can be chained together. For instance, to reverse the characters in a
+string one might want an iterator of characters and then a reverse iterator, like so
+
+```rust
+let rev_chars = "Hello, World!".chars().rev();
+```
+
+But `rev_chars` does not give us a `String` with the characters of "Hello, World!" reversed. It gives us an `impl Iterator<Item = char>`. It's a
+type which is _prepared_ to yield a sequence of characters, but it doesn't know how to finally assemble them. In this case it doesn't know if we
+want a `Vec<char>` or a `String`.
+
+Even though iterator adaptors do so much, they are _lazy_. But this only means that they are like an arrow drawn in a bow. They are not released
+until they have a target, called a [consuming adaptor][consuming adaptor], which uses up either a single iterator or the last iterator in a chain
+of iterators.
+
+[`collect`][collect] is one such consuming adaptor, and is what can be used to tell `rev_chars` what type of collection we want to end up with. If
+we want a `Vec<char>` we could insert the type into our call to `collect` like so
+
+```rust
+    let rev_chars = "Hello, World!".chars().rev().collect::<Vec<char>>();
+// or, since the collection can only be made of chars if it's a Vec, we don't have to specify them and just use
+// let rev_chars = "Hello, World!".chars().rev().collect::<Vec<_>>();
+```
+
+and `rev_chars` will now be a `Vec<char>`, but the double colons and outer angle brackets (also known as the `turbofish`) which passes the type
+information to the `collect` function is noisier syntax than we need. By specifying the type on the binding itself the `collect` can infer what
+collection it should create without the syntactical noise.
+
+```rust
+    let rev_chars: Vec<char> = "Hello, World!".chars().rev().collect();
+// or
+// let rev_chars: Vec<_> = "Hello, World!".chars().rev().collect();
+```
+
+If we wanted a `String` instead we could do it like so
+
+```rust
+    let rev_chars: String = "Hello, World!".chars().rev().collect();
+```
+
+But what if we want to return a collection from a function instead of letting a binding? We could do
+
+```rust
+fn get_reversed_chars(phrase: &str) -> Vec<char> {
+    phrase.chars().rev().collect()
+}
+```
+
+The return type of the `get_reversed_chars` function lets the `collect` function infer what collection to return. Here, though, we must specify the
+`Vec<char>` and not `Vec<_>` or we get the compilation error
+
+```rust
+// the type placeholder `_` is not allowed within types on item signatures
+// not allowed in type signatures
+// replace with the correct return type
+```
+
+Although you can sometimes use `collect` without specifying the type, at other times the compiler will state it requires the type for a `collect`, so it
+can be a good habit to be aware of how to specify what type you want to collect.
```

Comment:
Also not sure this is essential. Rustc will complain if it can't infer something, and students' natural reluctance to add a bunch of syntax will encourage them to discover the minimal limits of what rustc will accept.

Reply:
I agree. I basically put it in as a kind of closing thought to keep the ending from seeming abrupt, but I don't feel strongly about it.